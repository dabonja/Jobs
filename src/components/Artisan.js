import React from 'react';
import mechanic from '../icons/mechanic.png';
import { Link } from 'react-router-dom';

const Artisan = ({ name, company, rated, profession, location, id, getId }) => {

    return (
        <div className=" list-inline-item mt-3 shadow p-3 mb-5 bg-body rounded">
            <div id={id} className="card" style={{ width: '18rem', backgroundColor: '#d9ffea' }}>
                <img src={mechanic} className="card-img-top" alt="..." />
                <div className="card-body">
                    <p className="h5 fst-italic"> {name}</p>
                    <p className="card-text">Location: {location}</p>
                </div>
                <ul className="list-group list-group-flush" >
                    <li className="list-group-item">Rated: {rated === null ? 'Not rated' : rated}</li>
                    <li className="list-group-item">Company: {company}</li>
                    <li className="list-group-item">Profession: {profession}</li>
                </ul>
                <div className="card-body">
                    <a href="#" className="card-link">Card link</a>
                    <Link className="card-link" to="/ArtisanDetails" onClick={(e) => { getId(e) }} >Contact</Link>
                </div>
            </div>
        </div>
    )
}

export default Artisan;