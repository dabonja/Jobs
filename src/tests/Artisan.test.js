import Artisan from '../components/Artisan';;
import React from 'react';
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom';



test('Artisan component renders correctly',()=>{
    const tree = renderer.create(
        <Router>
        <Artisan name="Danijel"  company="Elektrotim" rated="7" profession="Serviser" location="Beska" id="1" getId={1} />
        </Router>
    ).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
});


