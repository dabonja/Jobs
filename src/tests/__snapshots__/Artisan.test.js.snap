// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artisan component renders correctly 1`] = `
<div
  className=" list-inline-item mt-3 shadow p-3 mb-5 bg-body rounded"
>
  <div
    className="card"
    id="1"
    style={
      Object {
        "backgroundColor": "#d9ffea",
        "width": "18rem",
      }
    }
  >
    <img
      alt="..."
      className="card-img-top"
      src="mechanic.png"
    />
    <div
      className="card-body"
    >
      <p
        className="h5 fst-italic"
      >
         
        Danijel
      </p>
      <p
        className="card-text"
      >
        Location: 
        Beska
      </p>
    </div>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        Rated: 
        7
      </li>
      <li
        className="list-group-item"
      >
        Company: 
        Elektrotim
      </li>
      <li
        className="list-group-item"
      >
        Profession: 
        Serviser
      </li>
    </ul>
    <div
      className="card-body"
    >
      <a
        className="card-link"
        href="#"
      >
        Card link
      </a>
      <a
        className="card-link"
        href="/ArtisanDetails"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </div>
</div>
`;
